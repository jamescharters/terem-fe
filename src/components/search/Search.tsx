import React, { useEffect, useState } from "react";
import Styles from "./Search.module.scss";

export type SearchProps = {
    onSearch: (queryText: string) => void;
};

export const Search: React.FC<React.PropsWithChildren<SearchProps>> = ({ onSearch }) => {
    const [searchText, setSearchText] = useState("");

    const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        setSearchText(event.target.value);
    };

    useEffect(() => onSearch(searchText), [onSearch, searchText]);

    return (
        <div className={Styles.terem__search}>
            <input type="search" onChange={onSearchChange} value={searchText} placeholder="Search for..." data-testid="input-search" />
        </div>
    );
};

export default Search;
