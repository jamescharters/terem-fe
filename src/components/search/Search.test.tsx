import { render, screen, waitFor  } from "@testing-library/react";
import { vi, describe, it, expect } from "vitest";
import Search from "./Search";
import userEvent from "@testing-library/user-event";

describe("Search", () => {
    it("renders correctly", () => {
        const result = render(<Search onSearch={() => {}} />);
        expect(result.asFragment()).toMatchSnapshot();
    });

    it("calls 'onSearch' when text is changed", async () => {
        const mockFn = vi.fn()
        render(<Search onSearch={mockFn} />);
        const input = await screen.findByTestId("input-search");
        userEvent.type(input, "new value");
        await waitFor(() => expect(mockFn).toHaveBeenCalledWith("new value"));
    });
});
