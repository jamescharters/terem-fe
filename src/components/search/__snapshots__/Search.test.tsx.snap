// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="_terem__search_6ac984"
  >
    <input
      data-testid="input-search"
      placeholder="Search for..."
      type="search"
      value=""
    />
  </div>
</DocumentFragment>
`;
